src/uu/ls/src/ls.rs

```diff
@@ -1084,8 +1079,9 @@ impl Config {
         };
 
         let dired = options.get_flag(options::DIRED);
-        if dired && format != Format::Long {
-            return Err(Box::new(LsError::ConflictingArgumentDired));
+        if dired {
+            // --dired implies --format=long
+            format = Format::Long;
         }
         if dired && format == Format::Long && options.get_flag(options::ZERO) {
```

Comment:
With the changes on lines 1082-1084 you no longer have to check `format`:
```suggestion
        if dired && options.get_flag(options::ZERO) {
```